src/core/xlio_extra.h

```diff
@@ -271,7 +271,7 @@ typedef enum {
     RING_LOGIC_PER_THREAD = 20, //!< RING_LOGIC_PER_THREAD
     RING_LOGIC_PER_CORE = 30, //!< RING_LOGIC_PER_CORE
     RING_LOGIC_PER_CORE_ATTACH_THREADS = 31, //!< RING_LOGIC_PER_CORE_ATTACH_THREADS
-    RING_LOGIC_PER_OBJECT = 32, //!< RING_LOGIC_PER_OBJECT
+    RING_LOGIC_NEIGH = 32, //!< RING_LOGIC_NEIGH
     RING_LOGIC_ISOLATE = 33, //!< RING_LOGIC_ISOLATE
     RING_LOGIC_LAST //!< RING_LOGIC_LAST
```

Comment:
[suggestion]
xlio_extra.h:
```C
enum {
    ...
    RING_LOGIC_PER_CORE_ATTACH_THREADS = 31,
    RING_LOGIC_LAST = 100
};
```
ring_allocation_logic.h:
```C
/* Internal ring allocation logic which is hidden from user. */
enum {
    RING_LOGIC_NEIGH = RING_LOGIC_LAST + 1,
    RING_LOGIC_ISOLATE,
};
```
ring_allocation_logic.cpp:
```C
bool is_logic_support_migration()
{
...
    m_res_key.get_ring_alloc_logic() < **RING_LOGIC_LAST**
}
```

Reply:
I am not sure about this change. It will create confusing conversions between ring_logic_t and ring_logic_public_t.
If we have to hide internal logics from the user of xlio_extra maybe it is better:
move ring_logic_t to ring_allocation_logic.h and make 
enum {
     ...
    RING_LOGIC_PER_CORE_ATTACH_THREADS = 31
} ring_logic_public_t

just for extra_api.h usage. The negative of this approach is that we have kind of duplicate definitions.
The best would be deriving one enum from another, but this is not possible in CPP.

Reply:
this is extra API and will be replaced by storage api with the groups. Depending on further steps, we may even remove all the obsolete extra API. So lets keep it as is and improve once we do cleanup.